### PR TITLE
Improved docs for std.conv.parse (part 2)

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -3089,11 +3089,20 @@ Target parse(Target, Source)(ref Source p)
...
+
+Returns:
+    A character of type `Target`
+
+Throws:
+    A $(LREF ConvException) if the range is empty.
  */
 Target parse(Target, Source)(ref Source s)
-    if (isExactSomeString!Source &&
+    if (isSomeString!Source && !is(Source == enum)
```
Comment (JackStouffer) [inline std/conv.d]: Private templates should never appear in documentation

File: std/conv.d [hunk 2]
```diff
@@ -3089,11 +3089,20 @@ Target parse(Target, Source)(ref Source p)
...
+
+Returns:
+    A character of type `Target`
+
+Throws:
+    A $(LREF ConvException) if the range is empty.
  */
 Target parse(Target, Source)(ref Source s)
-    if (isExactSomeString!Source &&
+    if (isSomeString!Source && !is(Source == enum) &&
```
Comment (JackStouffer) [inline std/conv.d]: Private templates should never appear in documentation

**Comment by schveiguy** [comment]
When I search for isExactSomeString on new doc page, still have 3 instances of it.

**Comment by JackStouffer** [comment]
I'm fixing this piece meal. See https://github.com/dlang/phobos/pull/4600

**Comment by schveiguy** [comment]
Piecemeal how? If your point is to rid the docs of this private template constraint, you should do that.

**Comment by JackStouffer** [comment]
I'm going function by function. All of the docs for `parse` suck.

File: std/conv.d (hunk 2 above)
Comment (WalterBright) [inline std/conv.d]: Adding another constraint looks like a behavior change, not just documentation. What does that have to do with being private?

**Comment by JackStouffer** [comment]
@WalterBright `isExactSomeString` is a private template defined in std.conv. It's declaration is the following:

    template isExactSomeString(T)
    {
        enum isExactSomeString = isSomeString!T && !is(T == enum);
    }

So I didn't add another constraint.

The problem that private templates in constraints have is that the user has no idea what they are and what they require. Using only the constraints that are documented in public functions makes it so the user knows exactly what's needed for the function to compile.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**